packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -1960,18 +1960,39 @@ export class ResourceService {
   }
 
   async setOwner(
-    resource: Resource,
-    ownershipType: EnumOwnershipType,
-    ownerId: string
+    resourceId: string,
+    userId?: string,
+    teamId?: string
   ): Promise<Ownership> {
+    if (isEmpty(userId) && isEmpty(teamId))
+      throw new AmplicationError("ownerId does not provide");
+
+    const ownershipType: EnumOwnershipType = userId
+      ? EnumOwnershipType.User
+      : EnumOwnershipType.Team;
+
+    const ownerId = userId ? userId : teamId;
+
+    const resource = await this.resource({ where: { id: resourceId } });
+
     if (resource.ownershipId) {
       return this.ownershipService.updateOwnership(
         resource.ownershipId,
         ownershipType,
         ownerId
       );
     } else {
-      return this.ownershipService.createOwnership(ownershipType, ownerId);
+      const ownerShip = await this.ownershipService.createOwnership(
+        ownershipType,
+        ownerId
+      );
+
+      await this.updateResource({
+        where: { id: resourceId },
+        data: { ownershipId: ownerShip.id },
+      });
```

Comment:
since we are already in the resource service, it is better to directly update the table instead of using the general update function. 
we do not want to expose the option to set the owner from any other entry point